app/controllers/incoming_controller.rb

```diff
@@ -0,0 +1,20 @@
+class IncomingController < ApplicationController
+  skip_before_action :verify_authenticity_token, only: [:create]
+
+  def create
+    puts "INCOMING PARAMS HERE: #{params}"
+
+    @user = User.find_by(email: params[:sender])
+
+    if @user.present?
+      @topic = Topic.find_or_create_by(user_id: @user.id, title: params[:subject])
+      @topic.save
```

Comment:
Line 11 `@topic.save` can be safely removed. Line 10 will `create` if needed, which does the save to the database.